app/src/main/java/app/gamenative/ui/screen/xserver/XServerScreen.kt

```diff
@@ -313,13 +418,12 @@ fun XServerScreen(
 
             var handled = false
             if (isGamepad) {
-                if (emulateKeyboardMouse) {
-                    handled = PluviaApp.inputControlsView?.onKeyEvent(it.event) == true
-                    if (!handled) handled = xServerView!!.getxServer().winHandler.onKeyEvent(it.event)
-                } else {
-                    handled = xServerView!!.getxServer().winHandler.onKeyEvent(it.event)
-                }
-                // handled = ExternalController.onKeyEvent(xServer.winHandler, it.event)
+                // Use standalone handler first (works independently of view visibility)
+                handled = physicalControllerHandler?.onKeyEvent(it.event) == true
```

Comment:
Not sure I like the idea of creating a physicalControllerHandler - why can't we extend inputControlsView?